src/main/java/io/beanmapper/dynclass/DynamicClassGeneratorHelper.java

```diff
@@ -0,0 +1,56 @@
+package io.beanmapper.dynclass;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
+import javassist.CtClass;
+
+public class DynamicClassGeneratorHelper {
+
+    private DynamicClassGeneratorHelper() {}
+
+    /**
+     *
+     * @param ctClass
+     * @return
+     */
+    public static Class<?> getNeighbourClass(CtClass ctClass) {
+
+        Class<?> neighbourClass = null;
+
+        try (InputStream inputStream = ClassLoader.getSystemResourceAsStream(
+                ctClass.getPackageName().replaceAll("[.]", "/"))) {
+            try (InputStreamReader isReader = new InputStreamReader(inputStream)) {
+                try (BufferedReader bufferedReader = new BufferedReader(isReader)) {
```

Comment:
Ligt er een beetje aan wat voor error message(s) je wilt gooien, maar anders zou je het ook plat kunnen slaan in 1 try catch.
![image](https://user-images.githubusercontent.com/105443742/188850798-d78299f5-cfa5-437f-9dc6-370577189f11.png)

Reply:
> > Ligt er een beetje aan wat voor error message(s) je wilt gooien, maar anders zou je het ook plat kunnen slaan in 1 try catch.
> > ![image](https://user-images.githubusercontent.com/105443742/188871560-4bbcd1ad-aef2-427e-ae5a-ff96c4a8e5c0.png)
> 
> Hmm, volgens mij is de geneste try-catch structuur nodig omdat het alle 3 try-with-resources statements zijn. Daarbij worden de InputStreamReader en BufferedReader automatisch geclosed, ook als er een error ontstaat.

Dat is inderdaad waarom ik het in de eerste instantie zo geschreven heb. Wist alleen niet zeker of het werkelijk vereist is om de bovenliggende readers ook te sluiten, wanneer de onderliggende InputStream al gesloten is. Na wat meer onderzoek blijkt dat wel noodzakelijk.

Dus nu is het de vraag of het gewenst is om een nested try-with blocks te hebben, of een enkele try with, waarin de de BufferedReader in een keer wordt geïnitialiseerd. Ik geef dan zelf de voorkeur aan nested try-with blocks. Wat denken jullie?